app/features/layout/AppErrorBoundary.tsx

```diff
@@ -1,23 +1,41 @@
+import { useCatch } from "@remix-run/react";
+import type { ApiResult } from "~/common/request.server";
 import { ErrorContainer } from "~/ui-toolkit/components/ErrorContainer/ErrorContainer";
 
-export const AppErrorBoundary = ({ error }) => {
-  let message = "Unknown error";
-
-  // It's a string
-  if (typeof error === "string") message = error;
-  // It's a normal error
-  else if (error?.message) message = error?.message;
-  // It's an array of errors
-  else if (Array.isArray(error)) {
-    message = error
-      ?.map((e) => e?.message)
-      .filter(Boolean)
-      .join(", ");
-  }
+// We can reuse this component for
+// both unhandled exceptions (ErrorBoundary)
+// and thrown Responses (CatchBoundary)
+export const AppErrorBoundary = ({ error: errorBoundaryError }) => {
+  let caughtThing = useCatch();
+  let error = caughtThing || errorBoundaryError;
```

Comment:
## Errors Step 2 
Picking back up from the other comment about throwing a Response.  By checking for either the `error` prop or the `useCatch()` result we can use this same component for BOTH the `CatchBoundary` and `ErrorBoundary`.  

We could alway choose not to for specific scenarios, but I like this as the default